docs/Boxes.md

```diff
@@ -0,0 +1,140 @@
+
+## The `Boxes` module {#sec-boxes}
+
+A _Box_ datatype is one whose values can hold either a success indication, optionally along with a value, or a failure indication, optionally with error information.
```

Comment:
Your definition of a box datatype only matches a small subset of examples of monads/wrappers: essentially variations of the option/error monad. Other examples, such as the Giry monad, the Reader monad, the Either monad, etc. (currently not part of the library) will not be covered by this description.

Do you suggest that we have a separate module for the other examples? If that is the case, I propose we use a less generic name than "box" for this module, and instead choose something more specific which indicates the "error handling" nature of the examples.

Reply:
Still thinking about an appropriate name.

Reply:
If you are inclined to separate each type into its own module, OCaml-style, now is the time while you're reorganizing anyway. It will definitely be a fair pain for consumers, but better one big pain than multiple smaller pains that add up to more overall pain. :)

Reply:
I would - but they depend on each other